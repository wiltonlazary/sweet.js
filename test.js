return function foo () {} /42/i
