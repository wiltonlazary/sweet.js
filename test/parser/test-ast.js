import test from 'ava';
import { getAst } from '../assertions';

test('does not include the lang directive in the AST', t => {
  t.snapshot(
    getAst(`
    'lang sweet.js';
  `)
  );
});

test('does include the use strict directive in the AST', t => {
  t.snapshot(
    getAst(`
    'use strict';
  `)
  );
});

test('includes export in AST', t => {
  t.snapshot(
    getAst(`
    export { b }
    `)
  );
});

test('includes export with renaming in AST', t => {
  t.snapshot(
    getAst(`
    export { b as c}
    `)
  );
});

test('includes export declaration in AST', t => {
  t.snapshot(
    getAst(`
    export var x = 1;
    `)
  );
});
