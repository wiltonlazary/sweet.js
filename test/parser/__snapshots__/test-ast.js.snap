// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does include the use strict directive in the AST 1`] = `
Module {
  "directives": Array [
    Object {
      "rawValue": "use strict",
      "type": "Directive",
    },
  ],
  "items": Array [],
  "loc": null,
  "type": "Module",
}
`;

exports[`does not include the lang directive in the AST 1`] = `
Module {
  "directives": Array [],
  "items": Array [],
  "loc": null,
  "type": "Module",
}
`;

exports[`includes export declaration in AST 1`] = `
Module {
  "directives": Array [],
  "items": Array [
    VariableDeclarationStatement {
      "declaration": VariableDeclaration {
        "declarators": Array [
          VariableDeclarator {
            "binding": BindingIdentifier {
              "loc": null,
              "name": "x_0",
              "type": "BindingIdentifier",
            },
            "init": LiteralNumericExpression {
              "loc": null,
              "type": "LiteralNumericExpression",
              "value": 1,
            },
            "loc": null,
            "type": "VariableDeclarator",
          },
        ],
        "kind": "var",
        "loc": null,
        "type": "VariableDeclaration",
      },
      "loc": null,
      "type": "VariableDeclarationStatement",
    },
    ExportFrom {
      "loc": null,
      "moduleSpecifier": null,
      "namedExports": Array [
        ExportSpecifier {
          "exportedName": "x",
          "loc": null,
          "name": "x_0",
          "type": "ExportSpecifier",
        },
      ],
      "type": "ExportFrom",
    },
  ],
  "loc": null,
  "type": "Module",
}
`;

exports[`includes export in AST 1`] = `
Module {
  "directives": Array [],
  "items": Array [
    ExportFrom {
      "loc": null,
      "moduleSpecifier": null,
      "namedExports": Array [
        ExportSpecifier {
          "exportedName": "b",
          "loc": null,
          "name": "b",
          "type": "ExportSpecifier",
        },
      ],
      "type": "ExportFrom",
    },
  ],
  "loc": null,
  "type": "Module",
}
`;

exports[`includes export with renaming in AST 1`] = `
Module {
  "directives": Array [],
  "items": Array [
    ExportFrom {
      "loc": null,
      "moduleSpecifier": null,
      "namedExports": Array [
        ExportSpecifier {
          "exportedName": "c",
          "loc": null,
          "name": "b",
          "type": "ExportSpecifier",
        },
      ],
      "type": "ExportFrom",
    },
  ],
  "loc": null,
  "type": "Module",
}
`;
